src/platform/tls_schannel.c

```diff
@@ -373,6 +384,13 @@ typedef struct QUIC_TLS {
 
 } QUIC_TLS;
 
+typedef struct QUIC_HP_KEY {
+    BCRYPT_KEY_HANDLE Key;
+    BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO* Info;
```

Comment:
Why can't this be on the stack instead?

Reply:
Having on the stack would incur performance cost (larger stack frame) for AES keys. This way, it's only a cost for ChaCha20-Poly1305 keys.

Reply:
```
typedef struct QUIC_HP_KEY {
    BCRYPT_KEY_HANDLE Key;
    QUIC_AEAD_TYPE Aead;
    BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO Info[0];
} QUIC_HP_KEY;
```